### ISSUE TITLE
Ameliorate the output messages of auto tactics

### ISSUE THREAD

**Issue by Ailrun** [Beluga-lang/Beluga#159]
The current output messages of auto tactics are not so clear and not providing much information to users. The following messages may still need some improvements.

- For `auto_intro`,
    - From
        ```
        ```
    - To
        ```
        Assumptions
          <A LIST OF THE ASSUMPTIONS>
        are automatically introduced for the subgoal of type
          <THE TYPE OF THE SUBGOAL>
        ```
- For `auto_solve_trivial`
    - From
        ```
        A goal <THE TYPE OF THE GOAL> is automatically solved.
        ```
    - To
        ```
        The subgoal
            <THE PATH TO THE SUBGOAL>
        of type
            <THE TYPE OF THE SUBGOAL>
        has been automatically solved by
            <THE EXPRESSION OF THE TYPE THAT SOLVES THE SUBGOAL>
        ```

**Comment by Ailrun** [comment]
`<THE EXPRESSION OF THE TYPE THAT SOLVES THE SUBGOAL>` is not always informative since it often uses newly introduced variables which have never been printed.

**Comment by Ailrun** [comment]
Printing `<A LIST OF THE ASSUMPTIONS>` only in `auto_intro` is not so easy since it is handled in the `intro` tactic.

**Comment by tsani** [comment]
I'm confused. I don't think one should remove manual intro since we can disable automation. Plus manual intro gives you the option to specify the names you want to use.
If you want the list of assumptions that are introduced, you can modify the core `intro'` function to return the list of names that were introduced, so `auto_intro` can use this to generate a nicer message.

**Comment by Ailrun** [comment]
@tsani I think that would be better. Thanks.

**Comment by Ailrun** [comment]
Resolved by 8c5765a9, bc09f24f, dcc71d36, 7340f610

**Comment by Ailrun** [comment]
Resolved by 8c5765a, bc09f24, dcc71d3, 7340f61
